src/passes/OptimizeInstructions.cpp

```diff
@@ -395,6 +395,25 @@ struct OptimizeInstructions
           return replaceCurrent(un);
         }
       }
+      {
+        // x < 0 || x > POSITIVE_CONST   ==>   x > POSITIVE_CONST (unsigned
+        // comparison)
+        Binary* bin;
+        Expression *x_left, *x_right;
+        Const* y;
+        if (matches(curr,
+                    binary(&bin,
+                           OrInt32,
+                           binary(LtSInt32, pure(&x_left), i32(0)),
+                           binary(GtSInt32, pure(&x_right), constant(&y)))) &&
+            y->type == Type::i32 && y->value.geti32() > 0 &&
+            ExpressionAnalyzer::equal(x_left, x_right)) {
```

Comment:
Rather than using `pure` + `::equal`, see `areConsecutiveInputsEqualAndFoldable` (they are consecutive aside from constants (which are not an issue), we need them to be equal, and we want to fold them together).

Reply:
Done